Refactor deploy documentation (#2502)
astronomer/docs: astro/deploy-dags.md

---
sidebar_label: 'Deploy DAGs to Astro'
title: 'Deploy DAGs'
id: deploy-dags
description: Learn about the different ways you can deploy code to Astro.
---

DAG-only deploys are the fastest way to deploy code to Astro. They are recommended if you only need to deploy changes made to the `dags` directory of your Astro project.

To push only DAGs to Astro for a faster development experience, you must enable the feature for each Deployment. You only need to enable the feature once. When it is enabled, you must still run `astro deploy` when you make a change to any file in your Astro project that is not in the `dags` directory.

Enabling DAG-only deploys on Astro has a few benefits:

- DAG-only deploys are significantly faster than running `astro deploy`.
- When you run `astro deploy --dags`, the workers and schedulers in your Deployment will immediately pick up your changes and not restart. This results in a more efficient use of running workers and no downtime for your Deployment.

> - You can have different sets of users deploy project changes versus DAG changes. See [DAG-based templates](https://docs.astronomer.io/astro/ci-cd-templates/template-overview#dag-based-templates) for how you can set this up in your CI/CD pipelines.

Author: The permissions would be at the level of the Git platform, rather than Astro. I think it's okay to keep this ambiguous

> - You can have different sets of users deploy project changes versus DAG changes. See [DAG-based templates](https://docs.astronomer.io/astro/ci-cd-templates/template-overview#dag-based-templates) for how you can set this up in your CI/CD pipelines.

Reviewer: Sorry, I disagree here. 
- Firstly, we are not clearly stating the "how" part causing it to be ambiguous.
- Secondly, permissions in GitHub actions are based on branch names and type of event (push, pull etc.) and not user-based.
an actual use-case can be "You can choose to have either DAG-only deploy or image-based deploys or a combination of both based on your Deployment environment in your CI/CD setup. This also allows you to control permissions around how the changes are pushed to your Dev or production environment."

- You can use DAG deploys to update your DAGs when you have slow upload speed on your internet connection.

## Enable DAG-only deploys on a Deployment

Before you enable DAG-only deploys on a Deployment, ensure that you have access to the Deployment's Astro project and can trigger deploys from your current machine with the Astro CLI. You cannot enable the DAG-only deploy feature in the Cloud UI.

1. Run the following command to enable the feature:

    ```sh
    astro deployment update --dag-deploy enable
    ```

2. When the prompt appears in the Astro CLI, select the Deployment where you want to enable the feature. Running tasks will not be interrupted, but new tasks will not be scheduled until you trigger your first DAG-only deploy.

> 3. Open your Deployment's Astro project.

Author: You don't need to open it before you run the first command right? I want to group steps together logically.

> 3. Open your Deployment's Astro project.

Reviewer: As a data engineer, I would rather start from an astro project as that is more logical to me when I think about an Airflow Deployment. For example, I would go the VS code (or directly to my Terminal/iTerm window) and open my Astro project and then run the commands from a Terminal window within VS code.

---

> sidebar_label: 'Deploy DAGs to Astro'
> title: 'Deploy DAGs'

Author: ```suggestion
sidebar_label: 'Deploy DAGs'
title: 'Deploy DAGs  to Astro'
```

3. Open your Deployment's Astro project.
4. Run the following command to finalize the setup and trigger a DAG-only deploy to your Deployment:  

    ```sh
    astro deploy --dags
    ```

    If you don't trigger a deploy after enabling the feature, your Deployment cannot schedule new tasks.
5. (Optional) Open your Deployment in the Cloud UI. Confirm your deploy was successful by checking the Deployment's **DAG Bundle Version**. The version name should include the date and time that you triggered the deploy.

## Trigger a DAG-only deploy

Triggering a DAG-only deploy pushes DAGs to Astro and mounts them to the workers and schedulers in your Deployment. DAG-only deploys do not disrupt running tasks and do not cause any components to restart when you push code. If you deploy changes to a DAG that is currently running, active task runs finish executing according to the code from before you triggered a deploy. New task runs are scheduled using the code from your latest deploy.

Run the following command to deploy only your `dags` directory to a Deployment:

```sh
astro deploy --dags
```

## Disable DAG-only deploys on a Deployment

> If you have Workspace Admin permissions, you can turn off DAG-only deploys for a Deployment at any time if your organization doesn't benefit from faster deploys or prefers a deployment method that is exclusively based on building and deploying your Astro project as a Docker image. When you turn off DAG-only deploys, the way Airflow and Astro read your DAGs changes, and all existing DAGs are removed from your Deployments and they might not appear in the Airflow UI.

Reviewer: ```suggestion
If you have Workspace Admin permissions, you can turn off DAG-only deploys for a Deployment at any time if your organization doesn't benefit from faster deploys or prefers a deployment method that is exclusively based on building and deploying your Astro project as a Docker image. 

Disabling DAG-only deploys is a two-step process which includes first disabling the feature and then deploying to your Deployment. This is because the way Airflow and Astro read your DAGs changes when you turn-off DAG-only deploys. 
```
mentioning that the DAGs disappear does not seem correct to me, rather we can just clearly state that it is a two step process.

## Enable DAG-only deploys on a Deployment

Before you enable DAG-only deploys on a Deployment, ensure that you have access to the Deployment's Astro project and can trigger deploys from your current machine with the Astro CLI. You cannot enable the DAG-only deploy feature in the Cloud UI.

1. Open your Deployment's Astro project.
2. Run the following command to enable the feature on your Deployment:

    ```sh
    astro deployment update --dag-deploy enable
    ```

3. When the prompt appears in the Astro CLI, select the Deployment where you want to enable the feature. Running tasks will not be interrupted, but new tasks will not be scheduled until you trigger your first DAG-only deploy.
4. Run the following command to finalize the setup and trigger a DAG-only deploy to your Deployment:  

    ```sh
    astro deploy --dags
    ```

    If you don't trigger a deploy after enabling the feature, your Deployment cannot schedule new tasks.

5. (Optional) Open your Deployment in the Cloud UI. Confirm your deploy was successful by checking the Deployment's **DAG Bundle Version**. The version name should include the date and time that you triggered the deploy.

> 
> 
> 

Author: ```suggestion
```
